ui/frontend/src/components/SignInForm.tsx

```diff
@@ -0,0 +1,113 @@
+import { useForm } from 'react-hook-form';
+import { Box, TextField, Button } from '@mui/material';
+import styled from '@emotion/styled';
+
+type SignInFormValues = {
+  username: string;
+  email: string;
+  password: string;
+};
+
+const SignInForm = () => {
+  const { register, handleSubmit, reset, clearErrors, formState } =
+    useForm<SignInFormValues>();
+  const { errors } = formState;
+
+  const onSubmit = (data: SignInFormValues) => {
+    console.log(data);
+
+    if (!Object.keys(errors).length) {
+      reset({ username: '', email: '', password: '' });
+      clearErrors();
+    }
+  };
+
+  return (
+    <Box
+      sx={{
```

Comment:
We'd probably be better using styled components with breakway code for the css located outside of the function